sherpa/astro/ui/tests/test_astro_ui_unit.py

```diff
@@ -1275,3 +1275,241 @@ def test_delete_bkg_model_with_bkgid(id, clean_astro_ui):
 
     emsg = 'background model 2 for data set {} has not been set'.format(id)
     assert str(exc.value) == emsg
+
+
+@requires_fits
+@pytest.mark.parametrize("loadfunc", [ui.load_grouping, ui.load_quality])
+def test_load_xxx_no_data(loadfunc, clean_astro_ui, tmp_path):
+    """What happens when there's no data??"""
+
+    path = tmp_path / 'data'
+    path.write_text('1\n0\n0\n')
+
+    with pytest.raises(IdentifierErr) as exc:
+        loadfunc(2, str(path))
+
+    assert str(exc.value) == 'data set 2 has not been set'
+
+
+@requires_fits
+@pytest.mark.parametrize("loadfunc", [ui.load_grouping, ui.load_quality])
+def test_load_xxx_not_pha(loadfunc, clean_astro_ui, tmp_path):
+    """What happens with a non-PHA dataset?"""
+
+    ui.load_arrays(2, [1, 2, 3], [2, 4, 9])
+
+    path = tmp_path / 'data'
+    path.write_text('1\n0\n0\n')
+
+    with pytest.raises(ArgumentErr) as exc:
+        loadfunc(2, str(path))
+
+    assert str(exc.value) == 'data set 2 does not contain PHA data'
+
+
+@requires_fits
+@pytest.mark.parametrize("idval", [None, 1, 'xx'])
+def test_load_grouping(idval, clean_astro_ui, tmp_path):
+    """Simple grouping check"""
+
+    x = [1, 2, 3]
+    y = [0, 4, 3]
+    if idval is None:
+        ui.load_arrays(1, x, y, ui.DataPHA)
+    else:
+        ui.load_arrays(idval, x, y, ui.DataPHA)
+
+    path = tmp_path / 'group.dat'
+    path.write_text('1\n-1\n1')
+
+    data = ui.get_data(idval)
+    assert data.grouping is None
+
+    if idval is None:
+        ui.load_grouping(str(path))
+    else:
+        ui.load_grouping(idval, str(path))
+
+    assert not data.grouped
```

Comment:
The documentation for `load_grouping` is explicit (but I, like you, had not realized it until my initial 'assert data.grouped' check failed).There is actually a reason for this, since it's really the combination of grouping and quality that defines the grouping, although I think we could argue that the quality values only affect `ignore_bad` calls.

You can think of it a bit like background subtraction - we don't automatically subtract the background, so we don't automatically group. However, this falls apart because we do automatically apply the grouping when the file is first read in (i.e. we get both quality and grouping set).

Long-answer-short - I think it's an edge case and we at least document the behavior, so I'm not in a hurry to change it. We can review it as part of the grouping and filtering work.

Reply:
>  I'm not in a hurry to change it. We can review it as part of the grouping and filtering work.
Agreed.